spec/services/carry_over_application_spec.rb

```diff
@@ -77,4 +76,68 @@ def original_application_form
+      FeatureFlag.activate(:restructured_work_history)
+    end
+
+    it 'carries over history and sets feature_structured_work_history to true' do
+      described_class.new(original_application_form).call
+      carried_over_application_form = ApplicationForm.last
+
+      expect(carried_over_application_form.application_work_experiences.count).to eq(2)
+      expect(carried_over_application_form.feature_restructured_work_history).to be(true)
+    end
+
+    it 'sets the work history section to incomplete' do
+      described_class.new(original_application_form).call
+      carried_over_application_form = ApplicationForm.last
+
+      expect(carried_over_application_form.work_history_completed).to be(false)
+    end
+
+    it 'only carries over required attributes' do
+      described_class.new(original_application_form).call
+
+      carried_over_application_form = ApplicationForm.last
+
+      carried_over_application_form.application_work_experiences.each do |experience|
+        expect(experience.working_pattern).to be_nil
+        expect(experience.working_with_children).to be_nil
+        expect(experience.role).to be_present
+        expect(experience.organisation).to be_present
+        expect(experience.details).to be_present
+      end
+    end
+
+    it 'infers that `currently_working` is false if there is no ongoing work history item' do
+      first_job = original_application_form.application_work_experiences.first
+      first_job.update(start_date: 4.years.ago, end_date: 3.years.ago)
+
+      second_job = original_application_form.application_work_experiences.last
+      second_job.update(start_date: 2.years.ago, end_date: 1.year.ago)
+
+      described_class.new(original_application_form.reload).call
+
+      carried_over_application_form = ApplicationForm.last
+      expect(carried_over_application_form.application_work_experiences.find_by(organisation: first_job.organisation).currently_working?).to be(false)
+      expect(carried_over_application_form.application_work_experiences.find_by(organisation: second_job.organisation).currently_working?).to be(false)
+    end
+
+    it 'infers that `currently_working` is true if there is an ongoing work history item' do
+      first_job = original_application_form.application_work_experiences.first
+      first_job.update(start_date: 3.years.ago, end_date: 2.years.ago)
+
+      second_job = original_application_form.application_work_experiences.last
+      second_job.update(start_date: 1.year.ago, end_date: nil)
+
+      described_class.new(original_application_form.reload).call
+
+      carried_over_application_form = ApplicationForm.last
+      expect(carried_over_application_form.application_work_experiences.find_by(organisation: first_job.organisation).currently_working?).to be(false)
+      expect(carried_over_application_form.application_work_experiences.find_by(organisation: second_job.organisation).currently_working?).to be(true)
```

Comment:
Spec still flaking in CI. Organisation is generated by Faker, I wonder if that's a case of duplicate secondary school being generated?

Perhaps worth retrieving these by start date?

Reply:
Thanks @malcolmbaig, will try that, I am a bit confused about this one because I've not seen it happen locally ever.